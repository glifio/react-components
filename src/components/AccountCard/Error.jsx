import styled from 'styled-components'
import { string, func } from 'prop-types'
import { ButtonV2 } from '../Button/V2'
import { StandardBox } from '../Layout'

const AccountErrorBox = styled(StandardBox)`
  display: flex;
  flex-direction: column;
  justify-content: space-between;
  width: 18em;
  height: 18em;
  background-color: var(--red-light);
  color: var(--red-dark);

  h3 {
    margin-top: 0;
  }

  hr {
    border-color: var(--red-dark);
  }

  button {
    color: var(--red-dark);
    border-color: var(--red-dark);

    &:hover {
      color: var(--white);
      border-color: var(--white);
      background-color: var(--red-light);
    }

    &:active {
      color: var(--red-light) !important;
      border-color: var(--white) !important;
      background-color: var(--white) !important;
    }
  }
`

export const AccountError = ({ errorMsg, onTryAgain }) => (
  <AccountErrorBox>
    <div>
      <h3>Error</h3>
      <hr />
      <p>{errorMsg}</p>
    </div>
    <ButtonV2 onClick={onTryAgain}>Try again</ButtonV2>
  </AccountErrorBox>
)

AccountError.propTypes = {
  onTryAgain: func.isRequired,
  errorMsg: string
}

AccountError.defaultProps = {
  errorMsg: ''
}
