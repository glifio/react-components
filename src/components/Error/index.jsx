import PropTypes from 'prop-types'
import Box from '../Box'
import Button from '../Button'
import Glyph from '../Glyph'
import OnboardCard from '../Card/OnboardCard'
import { SmartLink } from '../Link/SmartLink'
import { GLIF_DISCORD, GLIF_TWITTER } from '../../constants'

const ErrorView = ({ description, linkhref, linkDisplay, title, sendHome }) => {
  let sendHomeCB = sendHome
  if (!sendHome) {
    sendHomeCB = () => {
      window.location.href = window.location.origin
    }
  }
  return (
    <div>
      <OnboardCard
        bg='card.error.background'
        color='card.error.foreground'
        borderColor='card.error.background'
        style={{ wordBreak: 'break-word' }}
      >
        <Glyph color='status.fail.foreground' acronym='Er' />
        <h2>{title}</h2>
        <p style={{ fontSize: '1.125rem' }}>{description}</p>
        {linkhref && linkDisplay && (
          <p style={{ fontSize: '1.125rem' }}>
            <SmartLink href={linkhref}>{linkDisplay}</SmartLink>
          </p>
        )}
        <Box
          mt='1.5em'
          pt='1em'
          borderTop='1px solid'
          borderColor='card.error.foreground'
        >
          Get help in <SmartLink href={GLIF_DISCORD}>Discord</SmartLink> or hit
          us up on <SmartLink href={GLIF_TWITTER}>Twitter</SmartLink>
        </Box>
      </OnboardCard>
      <Box mt={5} textAlign='center'>
        <Button variant='secondary' title='Back' onClick={sendHomeCB} />
      </Box>
    </div>
  )
}

ErrorView.propTypes = {
  description: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  linkhref: PropTypes.string,
  linkDisplay: PropTypes.string,
  sendHome: PropTypes.func
}

export default ErrorView
