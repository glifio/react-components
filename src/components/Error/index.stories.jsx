import theme from '../theme'
import ThemeProvider from '../ThemeProvider'

import { ErrorView } from '.'

export default {
  title: 'Error/ErrorView',
  component: ErrorView,
  decorators: [
    Story => (
      <div style={{ display: 'flex', justifyContent: 'center' }}>
        <ThemeProvider theme={theme}>{Story()}</ThemeProvider>
      </div>
    )
  ],
  parameters: { actions: { argTypesRegex: '^on.*' } }
}

const Template = args => <ErrorView {...args} />

export const Base = Template.bind({})
Base.args = {
  title: 'Ledger only supports Chrome',
  description:
    'Please install Google Chrome to continue using your Ledger device, or choose an alternative setup option',
  linkDisplay: 'Install Google Chrome.',
  linkhref: 'https://www.google.com/chrome'
}
