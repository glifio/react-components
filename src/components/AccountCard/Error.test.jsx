import { render } from '@testing-library/react'
import { Base } from './Error.stories'

describe('AccountError', () => {
  test('renders the story', () => {
    const mockOnTryAgain = jest.fn()
    const { container } = render(
      <Base {...Base.args} onTryAgain={mockOnTryAgain} />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
