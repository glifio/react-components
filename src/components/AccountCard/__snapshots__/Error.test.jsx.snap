// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AccountError renders the story 1`] = `
.c3 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  text-align: center;
  white-space: nowrap;
  cursor: pointer;
  gap: 0.75rem;
  padding: 0.5em 0.75em;
  color: var(--black);
  border-width: 1px;
  border-style: solid;
  border-radius: 8px;
  border-color: var(--black);
  background: transparent;
}

.c3 svg path {
  stroke: var(--black);
}

.c3:hover {
  color: var(--purple-medium);
  border-color: var(--purple-medium);
}

.c3:hover svg path {
  stroke: var(--purple-medium);
}

.c3:active {
  color: var(--white) !important;
  border-color: var(--purple-medium) !important;
  background: var(--purple-medium) !important;
}

.c3:active svg path {
  stroke: var(--white) !important;
}

.c0 {
  padding: 1.5em;
  border-radius: 8px;
  overflow: hidden;
  text-align: center;
  word-break: break-word;
}

.c0 > *:first-child {
  margin-top: 0;
}

.c0 > *:last-child {
  margin-bottom: 0;
}

.c0 > svg:first-child {
  display: block;
  margin: 0 auto 1em;
}

.c0 > h2:first-child {
  margin: 0;
}

.c0 header:first-child {
  margin: -1.5em -1.5em 1.5em -1.5em;
  padding: 1.5em;
  background-color: var(--purple-medium);
  color: var(--white);
}

.c1 {
  background-color: var(--blue-gray);
}

.c1 hr {
  border-color: var(--gray-medium);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  width: 18em;
  height: 18em;
  background-color: var(--red-light);
  color: var(--red-dark);
}

.c2 h3 {
  margin-top: 0;
}

.c2 hr {
  border-color: var(--red-dark);
}

.c2 button {
  color: var(--red-dark);
  border-color: var(--red-dark);
}

.c2 button:hover {
  color: var(--white);
  border-color: var(--white);
  background-color: var(--red-light);
}

.c2 button:active {
  color: var(--red-light) !important;
  border-color: var(--white) !important;
  background-color: var(--white) !important;
}

<div
  class="c0 c1 c2"
>
  <div>
    <h3>
      Error
    </h3>
    <hr />
    <p>
      There was an error
    </p>
  </div>
  <button
    class="c3"
  >
    Try again
  </button>
</div>
`;
